synchronizer/l1_rollup_info_producer.go

```diff
@@ -327,24 +331,41 @@ func (l *l1RollupInfoProducer) onNewLastBlock(lastBlock uint64, launchWork bool)
 	return resp
 }
 
+func (l *l1RollupInfoProducer) canISendNewRequestsUnsafe() (bool, string) {
+	queued := l.filterToSendOrdererResultsToConsumer.numItemBlockedInQueue()
+	inChannel := len(l.outgoingChannel)
+	maximum := cap(l.outgoingChannel)
+	msg := fmt.Sprintf("inFilter:%d + inChannel:%d > maximum:%d?", queued, inChannel, maximum)
+	if queued+inChannel > maximum {
+		msg = msg + " ==> only allow retries"
+		return false, msg
+	}
+	msg = msg + " ==> allow new req"
```

Comment:
Why don't you just log the message here instead of returning the message and log it in other function??

Reply:
It's to have all info in the same log, a long the function accumulate the logs of this execution and at end it emmit only 1 info trace